hail/python/hail/methods/statgen.py

```diff
@@ -489,11 +530,11 @@ def process_block(block):
 
         # Processes one block group based on given idx. Returns a single struct.
         def process_y_group(idx):
-            X = hl.nd.array(block[entries_field_name].map(lambda row: mean_impute(select_array_indices(row, ht.kept_samples[idx])))).T
+            X = (hl.nd.array(block[entries_field_name].map(lambda row: mean_impute(select_array_indices(row, ht.kept_samples[idx])))) * ht.__sqrt_weight_nds[idx]).T
```

Comment:
do we think that this line will have any performance impact on the non-weighted case?

Reply:
I would think it would be minimal since it'll just broadcast out some multiply by 1s, but it'd be pretty easy to just add an if statement here and only do it if the user initially specified weights. I'll just do that.

Reply:
Addressed